include/RAJA/pattern/forall.hpp

```diff
@@ -252,10 +272,21 @@ inline namespace policy_by_value_interface
  *
  ******************************************************************************
  */
-template <typename ExecutionPolicy, typename IdxSet, typename LoopBody>
-RAJA_INLINE void forall_Icount(ExecutionPolicy&& p,
-                               IdxSet&& c,
-                               LoopBody&& loop_body)
+template <typename ExecutionPolicy, typename IdxSet, typename LoopBody, typename Res>
+RAJA_INLINE resources::EventProxy<Res> forall_Icount(ExecutionPolicy&& p,
```

Comment:
I don't see where Res is specified or deduced in this signature.
Do we test the policy_by_value_interface?

Reply:
Most of it gets tested, it appears the non resource policy_by_value_interface functions are being missed.

Reply:
I have changed some of the existing tests to ensure we are testing pass by value calls.